src/coreclr/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -3148,9 +3148,9 @@ HRESULT ProfToEEInterfaceImpl::GetAppDomainStaticAddress(ClassID classId,
         return E_INVALIDARG;
     }
 
-    // Some domains, like the system domain, aren't APP domains, and thus don't contain any
+    // The system domain isn't an APP domain and thus doesn't contain any
     // statics.  See if the profiler is trying to be naughty.
-    if (!((BaseDomain*) appDomainId)->IsAppDomain())
+    if (appDomainId == (AppDomainID)SystemDomain::System())
```

Comment:
How can the profiler ever get access to the `SystemDomain::System()` pointer? This looks like a unreachable code that can be deleted.

Reply:
I remember seeing some weirdness around the system domain in the profiler APIs. I think they do get a callback when the system domain is created, but it's been a while so I am checking now

Reply:
You are right: https://github.com/dotnet/runtime/blob/2ed43b6a8105563f491fb521d761e6eff6ac4448/src/coreclr/vm/appdomain.cpp#L1533-L1545

This should stay then.